core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -73,9 +73,13 @@ class NettyBlockTransferService(conf: SparkConf, securityManager: SecurityManage
       val server = transportContext.createServer(port, bootstraps.asJava)
       (server, server.getPort)
     }
-
     val portToTry = conf.getInt("spark.blockManager.port", 0)
-    Utils.startServiceOnPort(portToTry, startService, conf, getClass.getName)._1
+
+    if(Utils.hasPortRange(conf)){
+      Utils.startServiceFromPortRange(portToTry, startService, conf, getClass.getName)._1
```

Comment:
So we are leaking into Netty layer. This might make things little tricky. Is there a way we can avoid making changes in this layer?

To understand the logic, if spark.blockManager.port is 0 then startServiceFromPortRange will pick a port from a range?